src/biogeochem/CNCStateUpdate3Mod.F90

```diff
@@ -60,61 +66,79 @@ subroutine CStateUpdate3( num_soilc, filter_soilc, num_soilp, filter_soilp, &
             c = filter_soilc(fc)
             ! patch-level wood to column-level CWD (uncombusted wood)
-            cs_soil%decomp_cpools_vr_col(c,j,i_cwd) = cs_soil%decomp_cpools_vr_col(c,j,i_cwd) + &
+            if (.not. use_soil_matrixcn) then
+               cs_soil%decomp_cpools_vr_col(c,j,i_cwd) = cs_soil%decomp_cpools_vr_col(c,j,i_cwd) + &
                  cf_veg%fire_mortality_c_to_cwdc_col(c,j) * dt
 
             ! patch-level wood to column-level litter (uncombusted wood)
-            cs_soil%decomp_cpools_vr_col(c,j,i_met_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_met_lit) + &
+               cs_soil%decomp_cpools_vr_col(c,j,i_met_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_met_lit) + &
                  cf_veg%m_c_to_litr_met_fire_col(c,j)* dt
-            cs_soil%decomp_cpools_vr_col(c,j,i_cel_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_cel_lit) + &
+               cs_soil%decomp_cpools_vr_col(c,j,i_cel_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_cel_lit) + &
                  cf_veg%m_c_to_litr_cel_fire_col(c,j)* dt
-            cs_soil%decomp_cpools_vr_col(c,j,i_lig_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_lig_lit) + &
+               cs_soil%decomp_cpools_vr_col(c,j,i_lig_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_lig_lit) + &
                  cf_veg%m_c_to_litr_lig_fire_col(c,j)* dt
+            else
+            ! patch-level wood to column-level CWD (uncombusted wood)
+               cf_soil%matrix_Cinput%V(c,j+(i_cwd-1)*nlevdecomp) = cf_soil%matrix_Cinput%V(c,j+(i_cwd-1)*nlevdecomp) + &
+                 cf_veg%fire_mortality_c_to_cwdc_col(c,j) * dt
+
+            ! patch-level wood to column-level litter (uncombusted wood)
+               cf_soil%matrix_Cinput%V(c,j+(i_met_lit-1)*nlevdecomp) = cf_soil%matrix_Cinput%V(c,j+(i_met_lit-1)*nlevdecomp) + &
+                 cf_veg%m_c_to_litr_met_fire_col(c,j)* dt
+               cf_soil%matrix_Cinput%V(c,j+(i_cel_lit-1)*nlevdecomp) = cf_soil%matrix_Cinput%V(c,j+(i_cel_lit-1)*nlevdecomp) + &
+                 cf_veg%m_c_to_litr_cel_fire_col(c,j)* dt
+               cf_soil%matrix_Cinput%V(c,j+(i_lig_lit-1)*nlevdecomp) = cf_soil%matrix_Cinput%V(c,j+(i_lig_lit-1)*nlevdecomp) + &
+                 cf_veg%m_c_to_litr_lig_fire_col(c,j)* dt
+            end if
          end do
       end do
 
       ! litter and CWD losses to fire
-      do l = 1, ndecomp_pools
-         do j = 1, nlevdecomp
-            do fc = 1,num_soilc
-               c = filter_soilc(fc)
-               cs_soil%decomp_cpools_vr_col(c,j,l) = cs_soil%decomp_cpools_vr_col(c,j,l) - &
+      if(.not. use_soil_matrixcn)then
+         do l = 1, ndecomp_pools
+            do j = 1, nlevdecomp
+               do fc = 1,num_soilc
+                  c = filter_soilc(fc)
+!               if (.not. use_soil_matrixcn)then
+                  cs_soil%decomp_cpools_vr_col(c,j,l) = cs_soil%decomp_cpools_vr_col(c,j,l) - &
                     cf_veg%m_decomp_cpools_to_fire_vr_col(c,j,l) * dt
+!               end if
```

Comment:
Please remove the commented out line and line 102.

Reply:
done